src/pretix/plugins/banktransfer/tasks.py

```diff
@@ -366,22 +376,35 @@ def process_banktransfers(self, job: int, data: list) -> None:
                 transactions = _get_unknown_transactions(job, data, **job.owner_kwargs)
 
                 # Match order codes
+                regex_match_to_slug = {}
                 code_len_agg = Order.objects.filter(event__organizer=job.organizer).annotate(
                     clen=Length('code')
                 ).aggregate(min=Min('clen'), max=Max('clen'))
                 if job.event:
-                    prefixes = {job.event.slug.upper()}
+                    prefixes = {job.event.slug.upper(), job.event.slug.upper().replace("-", "")}
+                    if "-" in job.event.slug:
+                        regex_match_to_slug[job.event.slug.upper().replace("-", "")] = job.event.slug
                 else:
-                    prefixes = {e.slug.upper() for e in job.organizer.events.all()}
+                    prefixes = set()
+                    for e in job.organizer.events.all():
+                        prefixes.add(e.slug.upper())
+                        if "-" in e.slug:
+                            prefixes.add(e.slug.upper().replace("-", ""))
+                            regex_match_to_slug[e.slug.upper().replace("-", "")] = e.slug
 
                 # Match invoice numbers
                 inr_len_agg = Invoice.objects.filter(event__organizer=job.organizer).annotate(
                     clen=Length('invoice_no')
                 ).aggregate(min=Min('clen'), max=Max('clen'))
                 if job.event:
-                    prefixes |= {p.rstrip(' -') for p in Invoice.objects.filter(event=job.event).distinct().values_list('prefix', flat=True)}
+                    invoice_prefixes = Invoice.objects.filter(event=job.event).distinct().values_list('prefix', flat=True)
                 else:
-                    prefixes |= {p.rstrip(' -') for p in Invoice.objects.filter(event__organizer=job.organizer).distinct().values_list('prefix', flat=True)}
+                    invoice_prefixes = Invoice.objects.filter(event__organizer=job.organizer).distinct().values_list('prefix', flat=True)
+                for p in invoice_prefixes:
```

Comment:
Note the order_by so the DB really returns distinct values.
```suggestion
                    invoice_prefixes = Invoice.objects.filter(event__organizer=job.organizer)
                for p in invoice_prefixes.order_by().distinct().values_list('prefix', flat=True):
```